### DISCUSSION TITLE
Annotating throats at a certain location

### DISCUSSION THREAD

**Discussion by mourendong** [PMEAL/OpenPNM#2203] [Q&A] [answered]
Hi, everyone, 
   I try to mark the throat at a certain location, but I did not find a way to do it.
  For example, the length of the throat is 2.5e-10, and I want to mark the location of 1.0e-10, just like the picture I added manually (the black dot). 
![image](https://user-images.githubusercontent.com/47968460/144018856-4d011072-0031-4e2d-b7c1-b5712e635b7c.png)
Can you give me some advice on how to do it?

**Comment by jgostick** [comment, marked as answer]
You have a few options, but they all involve finding the [x,y,z] points you want to plot, so let's start there.  For each throat in the network you can find it's centroid (as the average of the neighbor) using ``crds = np.mean(pn.coords[pn.conns], axis=1)``

To plot them you can either (a) add them to your network (mangling your current network) (b) create a new network and plot them together or (c) update a custom version of our plot function to add them.  I think option b is the best bet:

``` python
pn = op.network.Cubic([5, 5, 1])
crds = np.mean(pn.coords[pn.conns], axis=1)
pn2 = op.network.GenericNetwork(Np=crds.shape[0], Nt=0)
pn2['pore.coords'] = crds
ax = op.topotools.plot_coordinates(pn, c='g')
ax = op.topotools.plot_connections(pn, ax=ax)
ax = op.topotools.plot_coordinates(pn2, ax=ax, c='r')
```

**Reply by mourendong** [reply]
Thanks for your kind reply!
The code as you write expresses the results, but I'm still a little confused.
As the example, the length of the throat is 2.5e-10, how to mark any location of the throat, if the point in the length of 1.0e-10, 0.8e-10, or 2.3e-10.
Maybe your reply involve it, I'm still a little confused about it.
Many thanks for your help!

**Reply by jgostick** [reply]
You'll need to alter the line: ``crds = np.mean(pn.coords[pn.conns], axis=1)`` to generate the coordinates you which to plot. I thought you wanted the 'mid-point' of the throat.  If you want some other point then you need to compute it as required.  The end result should be an ``Nt by 3`` array of [x,y,z] coordinates, which you can then use as ``'pore.coords'`` in a new fake network for the purposes of plotting.

**Reply by mourendong** [reply]
Many thanks for your kind explanation, I got it!
